Remove unnecessary boxing, since java1.5 these are not necessary

DanVanAtta opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/triplea-game/triplea/698)
<!-- Reviewable:end -->

ron-murhammer commented:
@DanVanAtta Looks like this needs rebased.

ron-murhammer commented:
@DanVanAtta Looks like the build failed on this one. Might just need to rerun it?

DanVanAtta commented:
I think it is time we @Ignored the failing test case
`570 tests completed, 1 failed, 5 skipped`

ron-murhammer commented:
@DanVanAtta Agree as long as its the same unit test that keeps sporadically failing (which I believe is the case).

RoiEXLab commented:
BTW... Side note for a future PR...
The IntegerMap object seems to be a mixture of a pre Java 1.5 utility object (it overloads the put method with Integer and int) and a calculating object which can multiply it's values by a specified amount... It's probably a good Idea to remove this object, and replace those calculation methods with static methods accepting a normal map object...

DanVanAtta commented:
@ron-murhammer  - Is the expectation to keep re-running the test until it passes? I am intentionally not re-running the single-unit test failure to give us a reminder that we need to fix those tests.

ron-murhammer commented:
@DanVanAtta In order to merge this PR as is, yes. If you want to fix it in this PR that's fine otherwise we should create an issue to track fixing/ignoring the test. We shouldn't be merging PRs that don't pass the automated build (unless they are actually build related and fixing something).

DanVanAtta commented:
The build of master is broken. We should use some context as to what our tools are telling us.  The green light we get is certainly not indication that everything is fine. We've had an issue about this for a long time: https://github.com/triplea-game/triplea/issues/274, with that problem around, the build of master is broken. It's a reality.

I think you should have went ahead and hit the re-run button @ron-murhammer  on this one. It's on the reviewers to bear the pain of a broken build process, so they are have incentive to fix it. It would also have spared some round trip times.

ron-murhammer commented:
@DanVanAtta And the issue has been minor enough for us to not fix for a long time but instead rerun builds. That is the stance we have adopted so far and unless we decide to either fix it or just ignore the automated build all together the standard is the PR must pass the checks. It is on the developer not reviewer to make sure it passes and addresses the issues if it doesn't, this is especially important in this project since we have a lack of reviewers. Not sure why you didn't either just rerun the build or add a commit to ignore the test sooner.

ron-murhammer commented on src/games/strategy/triplea/ai/proAI/ProBidAI.java:
```diff
@@ -63,17 +63,17 @@ public void bid(int PUsToSpend, final IPurchaseDelegate purchaseDelegate, final
+    final IntegerMap<ProductionRule> bestDefense = new IntegerMap<ProductionRule>();
+    final IntegerMap<ProductionRule> bestTransport = new IntegerMap<ProductionRule>();
+    final IntegerMap<ProductionRule> bestMaxUnits = new IntegerMap<ProductionRule>();
+    final IntegerMap<ProductionRule> bestMobileAttack = new IntegerMap<ProductionRule>();
```
Bad merge?

ron-murhammer commented on src/games/strategy/triplea/ai/proAI/ProBidAI.java:
```diff
@@ -169,7 +169,7 @@ public void bid(int PUsToSpend, final IPurchaseDelegate purchaseDelegate, final
                                                            // units, so we must be sure not to remove those)
     {
-      final List<ProductionRule> seaProductionRulesCopy = new ArrayList<>(seaProductionRules);
+      final List<ProductionRule> seaProductionRulesCopy = new ArrayList<ProductionRule>(seaProductionRules);
```
Bad merge?

ron-murhammer commented on src/games/strategy/triplea/ai/proAI/ProBidAI.java:
```diff
@@ -276,7 +276,7 @@ public void bid(int PUsToSpend, final IPurchaseDelegate purchaseDelegate, final
       }
     }
-    final List<ProductionRule> processRules = new ArrayList<>();
+    final List<ProductionRule> processRules = new ArrayList<ProductionRule>();
```
Bad merge?

ron-murhammer commented on src/games/strategy/triplea/ai/proAI/ProBidAI.java:
```diff
@@ -386,12 +386,12 @@ public void bidPlace(final IAbstractPlaceDelegate placeDelegate, final GameData
+    final CompositeMatch<Unit> enemyAttackUnit = new CompositeMatchAnd<Unit>(attackUnit, enemyUnit);
     // CompositeMatch<Unit> enemyTransUnit = new CompositeMatchAnd<Unit>(transUnit, enemyUnit);
-    final CompositeMatch<Unit> ourFactory = new CompositeMatchAnd<>(ownedUnit, Matches.UnitCanProduceUnits);
+    final CompositeMatch<Unit> ourFactory = new CompositeMatchAnd<Unit>(ownedUnit, Matches.UnitCanProduceUnits);
```
Bad merge?

ron-murhammer commented:
@DanVanAtta Looks like bad merges with the diamond notation updates. Pointed out the first several of them but there are a lot.

DanVanAtta commented:
> "@DanVanAtta And the issue has been minor enough for us to not fix for a long time but instead rerun builds."

No, it's been important for some time. The issue was brought up in the past, and instead of favoring  build stability we chose test coverage.  Now it's pretty apparent I believe that we should change that decision.


> "It is on the requester not reviewer to make sure it passes and addresses the issues if it doesn't"

You're assuming write access for the requester. We've established that requester will pretend as if they do not have write access when they submit PRs. Effectively dogfooding, and was the reason presented for why folks don't hit merge on their own PR. I also think that when we take turns seeing how this process feels from the other side (requester vs reviewer), we see process improvements. 

To have a complete process we need to accommodates non-write access PRs. For example, we've seen process gaps when we don't know how to assign PRs from @RoiEXLab . We should have a fast track only for building/reviewing/deploying code, and not a fast track only for those with write access.


> "this is especially important in this project since we have a lack of reviewers."

We have an overall lack of developers. We've been optimizing for reviewer time, which is good, but I think we need to have some balance and optimize a bit more for the submitter of code. For example, it's not worth pointing out every single small possible improvement and mistake, particularly when the big picture is being ignored, see the links at: https://github.com/triplea-game/triplea/issues/632, particularly: http://www.daedtech.com/how-to-use-a-code-review-to-execute-someones-soul/
